src/app/mod.rs

```diff
@@ -1227,6 +1230,32 @@ impl<'a, 'b> App<'a, 'b> {
         self.get_matches_from_safe(&mut env::args_os())
     }
 
+    /// Similar to [`App::get_matches`] but also reads args from
+    /// env-var `env_var_name`.
+    ///
+    /// Defaults to read from "VAR" if `env_var_name` is `None`.
+    pub fn get_matches_with_env(self, env_var_name: Option<String>)
+            -> ArgMatches<'a> {
+        let mut env_var = DEFAULT_ARGS_ENV_VAR.to_string();
+        if let Some(name) = env_var_name {
+            env_var = name;
+        }
+        // Build `args` by chaining `args_os()` and `var_os()`
+        let mut args: Vec<OsString> = env::args_os().collect();
+        // Handle first env-format:
+        // VAR="--option1=val1 --option2=val2" ./prog
+        if let Some(vargs) = env::var_os(&env_var) {
+            // TODO: Split on space below is ugly. How to split on all
+            //       utf-8-whitespace? Must use regex?
+            args = args.into_iter()
+                       .chain(vargs.split(0x20)
+                                   .into_iter()
+                                   .map(|s| s.to_os_string()))
+                       .collect();
```

Comment:
You can use `b' '` instead of `0x20`
`.split()` returns an iterator (`osstringext::OsSplit`), so `.into_iter()` does nothing.
`Vec` implements `std::iter::Extend` so this can be written as `args.extend(vargs.split(b' ').map(...))`.